control_protocol.md

```diff
@@ -0,0 +1,500 @@
+# Open Screen Control Protocol
+
+This document describes a control protocol suitable to support the
+implementation of
+the [Presentation API](https://w3c.github.io/presentation-api/) and
+the [Remote Playback API](https://w3c.github.io/remote-playback/).  It is
+intended to be implemented on top of one of the proposed Open Screen transport
+mechanisms such as [QUIC](quic.md) or [RTCDataChannel](datachannel.md).
+
+The control protocol is responsible for mapping Web application requests and
+data provided by Web applications onto a network transport for transmission
+between controlling user agents (controllers) and receiving user agents
+(presentations screens and remote playback devices).
+
+This document focuses on the syntax and semantics of the network messages
+themselves.  A future update will map these messages onto specific steps in the
+algorithms described in the respective specifications for the Presentation API
+and the Remote Playback API.
+
+## Requirements
+
+* The control protocol must implement
+  the [functional requirements for the Presentation API](requirements.md#presentation-api-requirements).
+* The control protocol must implement
+  the [functional requirements for the Remote Playback API](requirements.md#remote-playback-api-requirements).
+* The control protocol must meet [non-functional requirements](requirements.md#non-functional-requirements).
+
+If an [RTCDataChannel](datachannel.md) is used as the transport, additional
+messages will be required to exchange WebRTC signaling messages to establish the
+channel.
+
+## Message Transport
+
+To meet the requirements above, a message oriented control protocol is
+necessary.  That means that each party should be able to transmit a sequence of
+variable-length messages from the other, and have every message received by the
+other party, intact and in-order.
+
+The RTCDataChannel supports variable length messages by virtue of
+adopting [SCTP](https://tools.ietf.org/html/rfc4960) as its message transmission
+protocol. The
+[RTCDataChannel specification](https://tools.ietf.org/html/draft-ietf-rtcweb-data-channel-13)
+suggests there may be
+[implementation restrictions on message size](https://tools.ietf.org/html/draft-ietf-rtcweb-data-channel-13#section-4)
+and that implementations should
+[interleave messages](https://tools.ietf.org/html/draft-ietf-rtcweb-data-channel-13#section-6.6)
+for fairness.
+
+Meanwhile, QUIC is stream-oriented and not message oriented, so a message
```

Comment:
Since QUIC is stream-oriented, it'll have hand-of-line blocking issue while transmitting large message. Maybe we can introduce additional out-of-band channel, this can be discussed in another issue.

Reply:
QUIC is designed to prevent this from happening by multiplexing among streams between two endpoints.  The proposal submitted here suggests assigning one stream to each PresentationConnection, so a message sent on one connection shouldn't block a message sent on another.  This is something we can collect data on as well.